holopy/core/holopy_object.py

```diff
@@ -83,8 +80,15 @@ def _iteritems(self):
 
     @classmethod
     def to_yaml(cls, dumper, data):
-        return ordered_dump(dumper, '!{0}'.format(data.__class__.__name__), data)
-
+        tag = '!{0}'.format(data.__class__.__name__)
+        value = []
+        # heavily inspired by the source of PyYAML's represent_mapping
+        node = yaml.nodes.MappingNode(tag, value)
+        for key, item in data._iteritems():
+            node_key = dumper.represent_data(key)
+            node_value = dumper.represent_data(item)
+            value.append((node_key, node_value))
+        return node
```

Comment:
This is more of a question than a suggestion -- are there tests for these? If not it would be AWESOME to add some in, as if you TDD'd this with the refactor. Given that the yaml was causing some problems before, a few tests here would really help. 

If there are already tests, then ignore this please :)

And thanks for putting the ```ordered_dump``` function in the class method, which is the only place it's used.

Reply:
There is one minimal test of this to make sure the code runs. It's `test_yaml_output_of_serializable` found in test_io.py. I added to it a bit to make sure it has the correct behaviour about which attributes to include, but this still isn't sufficiently tested.